src/coreclr/tests/src/Interop/COM/Dynamic/Server/DispatchImpl.cpp

```diff
@@ -44,6 +44,9 @@ HRESULT DispatchImpl::DoGetTypeInfo(UINT iTInfo, ITypeInfo** ppTInfo)
     if (iTInfo != 0)
         return DISP_E_BADINDEX;
 
+    assert(_typeInfo != nullptr);
+
+    _typeInfo->AddRef();
```

Comment:
Instead of doing this the canonical approach of using `QueryInterface()` should be used to return a COM interface.

```c++
return _typeInfo->QueryInterface(__uuidof(*ppTInfo), (void**)ppTInfo);
```

Depending on compiler settings the following may also work.

```c++
return _typeInfo->QueryInterface(ppTInfo);
```